File: Python/secao04/Exercicio07.py
# var
p1 = 0
p2 = 0
p3 = 0
p4 = 0
contador_sit1 = 0
contador_sit2 = 0
contador_sit3 = 0
contador_sit4 = 0
quantidade = 0

# input
cod = int(input("Informe o código de identificação: "))

# processing
while cod != 0:
    print("------------------------------------------")
    print("Identifique o defeito:")
    print("1 - Necessita de esfera")
    print("2 - Necessita de limpeza")
    print("3 - Necessita de troca do cabo ou conector")
    print("4 - Quebrado ou inutilizado")
    defeito = int(input("Informe o defeito: "))
    print("==========================================")
    cod = int(input("Informe o código de identificação: "))

    if defeito == 1:
        contador_sit1 = contador_sit1 + 1
    elif defeito == 2:
        contador_sit2 = contador_sit2 + 1
    elif defeito == 3:
        contador_sit3 = contador_sit3 + 1
    elif defeito == 4:
        contador_sit4 = contador_sit4 + 1

    
    quantidade = quantidade + 1

p1 = (contador_sit1 / quantidade) * 100.0
p2 = (contador_sit2 / quantidade) * 100.0
p3 = (contador_sit3 / quantidade) * 100.0
p4 = (contador_sit4 / quantidade) * 100.0

print("============================================================================")
print("TOTAL DE APARELHOS: {0}".format(quantidade))
print("")
print(" Situação                                    ||  Quantidade  ||  Percentual")
print(" 1 - Necessita de esfera                     ||       {0}      ||     {1}".format(contador_sit1, p1))
print(" 2 - Necessita de limpeza                    ||       {0}      ||     {1}".format(contador_sit2, p2))
print(" 3 - Necessita de troca ou cabo do conector  ||       {0}      ||     {1}".format(contador_sit3, p3))
print(" 4 - Quebrado ou inutilizado                 ||       {0}      ||     {1}".format(contador_sit4, p4))

File: Java/src/secao04/Exercicio07.java
package secao07;

import java.util.Scanner;

public class Exercicio07 {
    public static void main(String[] args) {
        // var
        int identificacao, defeito, contador_total = 0, contador_sit_1 = 0, contador_sit_2 = 0, contador_sit_3 = 0,
                contador_sit_4 = 0;
        float p1, p2, p3, p4;
        Scanner teclado = new Scanner(System.in);

        // input
        System.out.println("Informe o código de identificação: ");
        identificacao = teclado.nextInt();

        // processing
        while (identificacao != 0) {
            System.out.println("");
            System.out.println("Informe o defeito:");
            System.out.println("========================================");
            System.out.println("1 - Necessita de esfera");
            System.out.println("2 - Necessita de limpeza");
            System.out.println("3 - Necessita troca de cabo ou conector");
            System.out.println("4 - Quebrado ou inutilizado");
            defeito = teclado.nextInt();

            if (defeito == 1) {
                contador_sit_1 = contador_sit_1 + 1;
            }
            if (defeito == 2) {
                contador_sit_2 = contador_sit_2 + 1;
            }
            if (defeito == 3) {
                contador_sit_3 = contador_sit_3 + 1;
            }
            if (defeito == 4) {
                contador_sit_4 = contador_sit_4 + 1;
            }

            System.out.println("Informe o código de identificação: ");
            identificacao = teclado.nextInt();
        }

        contador_total = (contador_sit_1 + contador_sit_2 + contador_sit_3 + contador_sit_4);
        p1 = ((float) contador_sit_1 / (float) contador_total) * (float) 100.00;
        p2 = ((float) contador_sit_2 / (float) contador_total) * (float) 100.00;
        p3 = ((float) contador_sit_3 / (float) contador_total) * (float) 100.00;
        p4 = ((float) contador_sit_4 / (float) contador_total) * (float) 100.00;

        // output
        System.out.printf("Quantidade de mouses: %d\n", contador_total);
        System.out.printf("Situação\t\t\t\t\t\t Total\t\t Porcentagem\n");
        System.out.printf("1 - Necessita de esfera\t\t\t\t\t %d\t\t %.2f\n", contador_sit_1, p1);
        System.out.printf("2 - Necessita de limpeza\t\t\t\t %d\t\t %.2f\n", contador_sit_2, p2);
        System.out.printf("3 - Necessita troca de cabo ou conector\t\t\t %d\t\t %.2f\n", contador_sit_3, p3);
        System.out.printf("4 - Quebrado ou inutilizado\t\t\t\t %d\t\t %.2f\n", contador_sit_4, p4);

        teclado.close();
    }
}
